ee/clickhouse/queries/trends/util.py

```diff
@@ -22,21 +22,31 @@
 }
 
 
-def process_math(entity: Entity) -> Tuple[str, str, Dict[str, str]]:
+def process_math(entity: Entity) -> Tuple[str, str, Dict[str, Any]]:
     aggregate_operation = "count(*)"
     join_condition = ""
-    params = {}
+    params: Dict[str, Any] = {}
     if entity.math == "dau":
         join_condition = EVENT_JOIN_PERSON_SQL
         aggregate_operation = "count(DISTINCT person_id)"
+    if entity.math == "unique_group":
+        if entity.math_group_type_index is None:
+            raise ValidationError(
+                {"math_group_type_index": "This field is required when `math` is set to unique_group."}, code="required"
+            )
+
+        aggregate_operation = f"count(DISTINCT JSONExtractString(properties, %(e_group_{entity.index}_math)s))"
+        params[f"e_group_{entity.index}_math"] = f"$group_{entity.math_group_type_index}"
```

Comment:
small nit: extract param out to a variable so we're sure there's no string interpolation mistakes here

Reply:
(and below)